src/System.Text.Json/tests/Serialization/DictionaryTests.KeyPolicy.cs

```diff
@@ -308,5 +311,94 @@ private class ClassWithExtensionDataProperty
            [JsonExtensionData]
             public Dictionary<string, object> Data { get; set; }
         }
+        
+        [Fact]
+        public static void CamelCaseSerialize_ForTypedDictionary_ApplyDictionaryKeyPolicy()
+        {
+            const string JsonCamel = @"{""keyDict"":{""Name"":""text"",""Number"":1000,""isValid"":true,""Values"":[1,2,3]}}";
+            var obj = new Dictionary<string, CustomClass>()
+            {
+                { "KeyDict", CreateCustomObject() }
+            };
+            var json = JsonSerializer.Serialize(obj, new JsonSerializerOptions()
+            {
+                DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
+            });
+
+            Assert.Equal(JsonCamel, json);
+        }
+
+        private class CustomClass
+        {
+            public string Name { get; set; }
+            public int Number { get; set; }
+            public bool isValid { get; set; }
+            public List<int> Values { get; set; }
+        }
+        
+        private static CustomClass CreateCustomObject()
+        {
+            return new CustomClass { Name = "text", Number = 1000, isValid = true, Values = new List<int>() { 1, 2, 3 } };
+        }
+
+        [Fact]
+        public static void CamelCaseSerialize_ForNestedTypedDictionary_ApplyDictionaryKeyPolicy()
+        {
+            const string JsonCamel = @"{""keyDict"":{""nestedKeyDict"":{""Name"":""text"",""Number"":1000,""isValid"":true,""Values"":[1,2,3]}}}";
+            var options = new JsonSerializerOptions
+            {
+                DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
+            };
+            var obj = new Dictionary<string, Dictionary<string, CustomClass>>(){
+                { "KeyDict", new  Dictionary<string,CustomClass>()
+                {{ "NestedKeyDict", CreateCustomObject() }}
+            }};
+            var json = JsonSerializer.Serialize(obj, new JsonSerializerOptions()
+            {
+                DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
+            });
+
+            Assert.Equal(JsonCamel, json);
+        }
+
+        private class TestClassWithDictionary
+        {
+           public Dictionary<string, CustomClass> Data { get; set; }
```

Comment:
A test where we serialize a class where we have both a dictionary and extension data (both with contents) would be good:

```c#
private class TestClassWithDictionaryAndExtensionData
{
    public Dictionary<string, CustomClass> Data { get; set; }

    [JsonExtensionData]
    public Dictionary<string, JsonElement> ExtensionData { get; set; }
}
```

Reply:
Any special reason to use JsonElement? 
In order to pass JsonElement values to the ExtensionData I would need to deserialize into a JsonDocument first, thing that I can avoid by using `Dictionary<string, object>` instead.